### PR TITLE
[SHEL-84]-Transaction-Route

### PR DESCRIPTION
**Pull request by wutpppppppppppppppppp**
This pull request introduces significant changes to the transaction handling in the project. It includes the addition of new transaction-related functionality, updates to existing files to integrate the new features, and some minor improvements.

### Transaction Handling:

* Added `TransactionResponse` struct in `internal/dto/transaction.go` to define the response format for transactions.
* Updated `Transaction` struct in `internal/entity/transaction.go` to include `Product` and restructured fields.
* Introduced `transactionRepository` in `internal/transaction/transaction.repository.go` to handle database operations for transactions.
* Created `transactionService` in `internal/transaction/transaction.service.go` to manage transaction-related HTTP requests.
* Added `transactionUsecase` in `internal/transaction/transaction.usecase.go` to implement business logic for transactions.

### Integration:

* Modified `internal/server/server.go` to include transaction routes and dependencies. [[1]](diffhunk://#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23R6) [[2]](diffhunk://#diff-d0706a4ae4ecf91c5224ca388f31a087a36af06c40dc81321da7958b25c70b23R29)

### Database:

* Updated `docker-compose.yml` to use PostgreSQL version 13 instead of the latest version.

### Minor Changes:

* Changed the output file extension to `.exe` in the `Makefile` for compatibility.
* Added `FindByID` method to `userRepository` in `internal/user/user.repository.go`.

### CODE REVIEW

File: internal/transaction/transaction.repository.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		Db: db,
+	}
+}
+
+func (r *transactionRepository) CreateTransaction(pctx context.Context, txn *entity.Transaction) error {
+	return r.transactionTable(pctx).Create(txn).Error
+}
+
+func (r *transactionRepository) GetTransactions(pctx context.Context) ([]*entity.Transaction, error) {
+	var txns []*entity.Transaction
```
Comment (TGRZiminiar) [inline internal/transaction/transaction.repository.go:40]: Here a variable is declared as a pointer, but in FindByTransactionId, a regular variable is used
Choose one approach have some consistency

File: internal/transaction/transaction.service.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			Success: false,
+		})
+	}
+	txn := entity.Transaction{
+		ProductID: uint(id),
+		BuyerID:   userId,
+	}
+	newTxn, err := s.transactionUsecase.CreateTransaction(ctx.UserContext(), &txn)
+	if err != nil {
+		switch err.Error() {
```
Comment (TGRZiminiar) [inline internal/transaction/transaction.service.go:52]: Verify the use of HTTP status codes I don't think Conflict or Not Found are being used correctly

File: internal/transaction/transaction.service.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	if err != nil {
+		return ctx.Status(fiber.StatusBadRequest).JSON(dto.HttpResponse{
+			Error:   "Invalid ID",
+			Success: false,
+		})
+	}
+
+	txn, err := s.transactionUsecase.GetTransactionByID(ctx.UserContext(), uint(id))
+	if err != nil {
+		return ctx.Status(fiber.StatusNotFound).JSON(dto.HttpResponse{
```
Comment (TGRZiminiar) [inline internal/transaction/transaction.service.go:111]: check status

File: internal/transaction/transaction.service.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+func (s *transactionService) GetTransactions(ctx *fiber.Ctx) error {
+	txns, err := s.transactionUsecase.GetTransactions(ctx.UserContext())
+	if err != nil {
+		return ctx.Status(fiber.StatusInternalServerError).JSON(dto.HttpResponse{
+			Error: err.Error(),
+		})
+	}
+
+	return ctx.Status(fiber.StatusOK).JSON(txns)
```
Comment (TGRZiminiar) [inline internal/transaction/transaction.service.go:96]: So some route use dto.HttpResponse and this route just send a data out like this? 
have some consistency

File: internal/transaction/transaction.service.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		})
+	}
+	txns, err := s.transactionUsecase.GetTransactionsByUserID(ctx.UserContext(), uint(id))
+	if err != nil {
+		return ctx.Status(fiber.StatusInternalServerError).JSON(dto.HttpResponse{
+			Error: err.Error(),
+		})
+	}
+
+	return ctx.Status(fiber.StatusOK).JSON(txns)
```
Comment (TGRZiminiar) [inline internal/transaction/transaction.service.go:135]: So some route use dto.HttpResponse and this route just send a data out like this?
have some consistency

**Review by TGRZiminiar — CHANGES_REQUESTED** [review]

File: internal/transaction/transaction.repository.go (hunk 1 above)
Comment (wutpppppppppppppppppp) [reply internal/transaction/transaction.repository.go:40]: the commit 0bdb69782875a0f7e1a2f41c87f294d782d5588d solved the issue...

File: internal/transaction/transaction.service.go (hunk 2 above)
Comment (wutpppppppppppppppppp) [reply internal/transaction/transaction.service.go:52]: f1c7cc53cb4367ec5429ee274dcdb7185a11aac6 would solve this

File: internal/transaction/transaction.service.go (hunk 3 above)
Comment (wutpppppppppppppppppp) [reply internal/transaction/transaction.service.go]: f1c7cc53cb4367ec5429ee274dcdb7185a11aac6

**Pull request merged**